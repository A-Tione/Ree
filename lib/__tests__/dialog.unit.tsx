import React, {useState} from 'react'
import {render} from "@testing-library/react";
import '@testing-library/jest-dom'
import Dialog, {alert, confirm, modal} from '../dialog/dialogs'

jest.mock('../icon/importIcons', () => {
  return {
    context: () => {
      const keys = () => [];
      const module = (key: string) => null;
      return {
        keys,
        module,
      };
    }
  };
});

interface WrapperComponentProps {
  initialVisible?: boolean;
  text: string;
}

describe('Dialog', () => {
  const WrapperComponent: React.FC<WrapperComponentProps> = ({ initialVisible = false, text }) => {
    const [x, setX] = useState(initialVisible);
    return (
      <Dialog
        visible={x}
        buttons={[
          <button onClick={() => setX(false)}>Confirm</button>,
          <button onClick={() => setX(false)}>Cancel</button>,
        ]}
        onClose={() => setX(false)}
      >
        <strong>{text}</strong>
      </Dialog>
    );
  };
    it('是个 dialog', () => {
      const { queryByText } = render(<WrapperComponent text={'hihi'} />);
      expect(queryByText('hihi')).toBeNull();
    })
    it('可以接受 visible', () => {
      const { queryByText } = render(<WrapperComponent initialVisible={true} text={'visible'} />);
      expect(queryByText('visible')).toBeInTheDocument();
    })
    // it('可以使用 alert', () => { 
    //     const close = alert('Alert')
    //     expect(close).toBeInstanceOf(Function)
    // })
    // it('可以使用 confirm', () => {
    //   const {queryByText} = render(confirm('Content', () => {}, () => {}))
    // })
    it('可以使用 modal', () => {
        const close = modal(<h1>你好</h1>, [<button onClick={() => close()}>close</button>])
        expect(close).toBeInstanceOf(Function)
    })
})