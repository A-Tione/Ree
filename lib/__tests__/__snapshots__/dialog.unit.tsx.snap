// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog 可以接受 visible 1`] = `<div />`;

exports[`Dialog 是个 dialog 1`] = `<div />`;
